app/controllers/search_controller.rb

```diff
@@ -34,6 +34,25 @@ def show
       raise Discourse::InvalidParameters.new("page parameter must not be greater than 10")
     end
 
+    if request.user_agent &&
+         CrawlerDetection.crawler?(request.user_agent, request.headers["HTTP_VIA"])
```

Comment:
We already do crawler detection for all requests in `ApplicationController`. I think we should be able to use that here instead of having to detect the crawler again.

https://github.com/discourse/discourse/blob/9025d98eaf2afc513f5400932dd6db73a2dfd917/app/controllers/application_controller.rb#L75